iconservice/icx/issue/regulator_storage.py

```diff
@@ -0,0 +1,91 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Optional
+
+from ...utils.msgpack_for_db import MsgPackForDB
+
+if TYPE_CHECKING:
+    from ...database.db import ContextDatabase
+    from ...iconscore.icon_score_context import IconScoreContext
+
+
+class RegulatorStorage(object):
+
+    _CURRENT_CALC_PERIOD_ISSUED_ICX_KEY = b'current_calc_period_issued_icx'
+    _PREV_CALC_PERIOD_ISSUED_ICX_KEY = b'prev_calc_period_issued_icx'
+    _OVER_ISSUED_ICX_KEY = b'over_issued_icx'
+    _OVER_ISSUED_I_SCORE_KEY = b'over_issued_i_score'
+
+    def __init__(self, db: 'ContextDatabase'):
+        self._db: 'ContextDatabase' = db
+
+    def close(self,
+              context: 'IconScoreContext') -> None:
+        """Close the embedded database.
+
+        :param context:
+        """
+        if self._db:
+            self._db.close(context)
+            self._db = None
+
+    def put_current_calc_period_issued_icx(self,
+                                           context: 'IconScoreContext',
+                                           current_calc_period_issued_amount: int):
+
+        encoded_current_issued_amount = MsgPackForDB.dumps(current_calc_period_issued_amount)
+        self._db.put(context, self._CURRENT_CALC_PERIOD_ISSUED_ICX_KEY, encoded_current_issued_amount)
+
+    def get_current_calc_period_issued_icx(self, context: 'IconScoreContext') -> int:
+        encoded_current_issued_amount = self._db.get(context, self._CURRENT_CALC_PERIOD_ISSUED_ICX_KEY)
+        current_issued_amount = 0
+        if encoded_current_issued_amount is not None:
+            current_issued_amount = MsgPackForDB.loads(encoded_current_issued_amount)
+        return current_issued_amount
+
+    def put_prev_calc_period_issued_icx(self, context: 'IconScoreContext', prev_calc_period_issued_amount: int):
+        encoded_prev_issued_amount = MsgPackForDB.dumps(prev_calc_period_issued_amount)
+        self._db.put(context, self._PREV_CALC_PERIOD_ISSUED_ICX_KEY, encoded_prev_issued_amount)
+
+    def get_prev_calc_period_issued_icx(self, context: 'IconScoreContext') -> Optional[int]:
+        encoded_prev_issued_amount = self._db.get(context, self._PREV_CALC_PERIOD_ISSUED_ICX_KEY)
+        prev_issued_amount: Optional[int] = None
+        if encoded_prev_issued_amount is not None:
+            prev_issued_amount = MsgPackForDB.loads(encoded_prev_issued_amount)
+        return prev_issued_amount
+
+    def put_over_issued_icx(self, context: 'IconScoreContext', over_issued_icx: int):
```

Comment:
I think its variable don't need to use.
redundant field

Reply:
I have reflected your suggestion.